### PR TITLE
Add SConsDeps generator

### PR DESCRIPTION
**Pull request by czoido**
Changelog: Feature: Add SConsDeps generator.
Docs: https://github.com/conan-io/docs/pull/3371

Related to: https://github.com/conan-io/conan/issues/14717

Functional tests only for Linux

### CODE REVIEW

File: conan/tools/scons/sconsdeps.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class SConsDeps:
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self._ordered_deps = []
+        self._generator_file = 'SConscript_conandeps'
+        check_using_build_profile(self._conanfile)
+
+    @property
+    def ordered_deps(self):
+        if not self._ordered_deps:
```
Comment (memsharded) [inline conan/tools/scons/sconsdeps.py]: This doesn't work when no deps are defined. Better initialize it with ``None`` in the constructor, and compare with ``if self._ordered_deps is None``

File: conan/tools/scons/sconsdeps.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            "FRAMEWORKPATH" : {{info.frameworkdirs or []}},
+            "CPPDEFINES"  : {{info.defines or []}},
+            "CXXFLAGS"    : {{info.cxxflags or []}},
+            "CCFLAGS"     : {{info.cflags or []}},
+            "SHLINKFLAGS" : {{info.sharedlinkflags or []}},
+            "LINKFLAGS"   : {{info.exelinkflags or []}},
+        },
+        {% if dep_version is not none %}"{{dep_name}}_version" : "{{dep_version}}",{% endif %}
+        """)
+        sections = ["conan = {\n"]
```
Comment (memsharded) [inline conan/tools/scons/sconsdeps.py]: Wasn't the section called ``conandeps``? (to account for a potential ``conantoolchain``)

**Review by memsharded — APPROVED** [review]
Looking good 👍

File: conan/tools/scons/sconsdeps.py (hunk 2 above)
Comment (czoido) [reply conan/tools/scons/sconsdeps.py]: Right now, as it is is this:

```
# SConscript_conandeps generated by conan
conan = {
        # this one aggregates all libs
        "conandeps" : {
            "CPPPATH" ...
        },
        
        # now separate libs
        "zlib" : {
            "CPPPATH" ...
        },
        "zlib_version" ...
        ...
```

loaded like this in the SConscript:

```
# SConscript by the consumer
conan = SConscript('./SConscript_conandeps')
flags = conan["conandeps"]
env.MergeFlags(flags)
```

Maybe we can do:

```
conandeps = {
        # this one aggregates all libs
        "conanlibs" : { # all_libs? aggregated_libs?...
            "CPPPATH" ...
        },
        ...
```

then:

```
conan = SConscript('./SConscript_conandeps')
flags = conandeps["conanlibs"]
env.MergeFlags(flags)
```

Something like that?

**Review by memsharded — APPROVED** [review]

**Pull request merged**